gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentBloomFilter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_GC_BLOOM_FILTER_FPP;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_GC_BLOOM_FILTER_SIZE;
+
+import com.google.common.hash.BloomFilter;
+import com.google.common.hash.Funnels;
+import java.io.Serializable;
+import java.util.Map;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+
+/** A utility class wrapping bloom filter functionality. */
+public class ContentBloomFilter implements Serializable {
+
+  private final long expectedEntries;
+  private final double fpp;
+  private static final long serialVersionUID = -693336833916979221L;
+
+  // track iceberg contents only using the snapshot id (long bloom filter)
+  // as the consumer of GC results needs only this info for clean up.
+  private volatile BloomFilter<Long> icebergTableFilter;
+  // track view contents only using the version-id (int bloom filter)
+  // as the consumer of GC results needs only this info for clean up.
+  private volatile BloomFilter<Integer> icebergViewFilter;
```

Comment:
The implementation only needs one BF, that uses the content-type+id. Having one BF per content type is not necessary, but wastes a lot of space. It also complicates this class with all the `if-else` chains.

Reply:
Sorry, I expect this design to be scalable when we add more content types like delta lake.
Having different bloom filter per content type will make sense for it. It may need binary filter or string filter.

Reply:
Also it is lazy initialization. If Iceberg view content type is not present for that reference, then that bloom filter will not be created.

Reply:
How is it "scaleable" when you have to touch this code and add even more if-else clauses, allocate more heap space etc etc?
A BF does not care about the "source type".
And further: the presence of a single content allocates memory for a BF for the total number of commits, which is a huge waste of resources.

Reply:
string bloom filter will be slow (as it is meant for variable size).  
is it ok ?